- Support symbol visibility with Clang
- Don't put linker flags in CXXFLAGS

--- CMakeLists.txt.orig	2014-10-02 14:05:46 UTC
+++ CMakeLists.txt
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 2.8.6)
+cmake_policy(SET CMP0043 OLD)
 project(liblastfm)
 
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
@@ -62,11 +63,11 @@ else()
 endif()
 
 
-if(CMAKE_COMPILER_IS_GNUCXX)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     add_definitions("-fno-operator-names -fvisibility-inlines-hidden -fvisibility=hidden")
 endif()
 if(UNIX AND NOT APPLE)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,--no-undefined")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
 endif()
 
 if(MSVC)
